compiler/kernel/op.go

```diff
@@ -52,16 +53,17 @@ import (
 var ErrJoinParents = errors.New("join requires two upstream parallel query paths")
 
 type Builder struct {
-	rctx         *runtime.Context
-	mctx         *super.Context
-	env          *exec.Environment
-	readers      []zio.Reader
-	progress     *zbuf.Progress
-	channels     map[string][]zbuf.Puller
-	deletes      *sync.Map
-	udfs         map[string]dag.Expr
-	compiledUDFs map[string]*expr.UDF
-	resetters    expr.Resetters
+	rctx                *runtime.Context
+	mctx                *super.Context
+	env                 *exec.Environment
+	readers             []zio.Reader
+	progress            *zbuf.Progress
+	channels            map[string][]zbuf.Puller
+	deletes             *sync.Map
+	udfs                map[string]*udf
```

Comment:
No need for type udf.  Just use dag.Func.

```suggestion
	udfs                map[string]*dag.Func
```